torch_audiomentations/augmentations/background_noise.py

```diff
@@ -0,0 +1,92 @@
+import math
+import numpy as np
+import random
+import soundfile
+import torch
+from ..core.transforms_interface import BasicTransform, EmptyPathException
+from ..utils.file import find_audio_files, load_audio
+from ..utils.dsp import calculate_rms, calculate_desired_noise_rms
+
+
+class ApplyBackgroundNoise(BasicTransform):
+    """
+    Applies a background noise to the input signal.
+    """
+
+    def __init__(
+        self, bg_path, min_snr_in_db=3, max_snr_in_db=30, device=torch.device("cpu"), p=0.5
+    ):
+        super(ApplyBackgroundNoise, self).__init__(p)
+        self.bg_path = find_audio_files(bg_path)
+
+        if len(self.bg_path) == 0:
+            raise EmptyPathException("There are no supported audio files found.")
+
+        self.min_snr_in_db = min_snr_in_db
+        self.max_snr_in_db = max_snr_in_db
+        self.device = device
+
+    def randomize_parameters(self, samples, sample_rate):
+        super(ApplyBackgroundNoise, self).randomize_parameters(samples, sample_rate)
+        if self.parameters["should_apply"] and samples.size(0) > 0:
+            bg_file_paths = random.choices(self.bg_path, k=samples.size(0))
+            bg_audios = []
+            for index, bg_file_path in enumerate(bg_file_paths):
+                bg_audio_info = soundfile.info(bg_file_path, verbose=True)
+                bg_audio_num_samples = math.ceil(
+                    sample_rate * bg_audio_info.frames / bg_audio_info.samplerate
+                )
+                samples_num_samples = samples[index].size(0)
+
+                # ensure that background noise has the same length as the sample
+                if bg_audio_num_samples < samples_num_samples:
+                    bg_start_index = 0
+                    bg_stop_index = bg_audio_info.frames
+                    loaded_bg_audio_num_samples = bg_audio_num_samples
+
+                    current_bg_audio = load_audio(
+                        bg_file_path,
+                        sample_rate=sample_rate,
+                        start=bg_start_index,
+                        stop=bg_stop_index,
+                    )
+                    bg_audio = [current_bg_audio]
+
+                    while loaded_bg_audio_num_samples < samples_num_samples:
+                        current_bg_audio = bg_audio[-1]
+                        loaded_bg_audio_num_samples += current_bg_audio.shape[0]
+                        bg_audio.append(current_bg_audio)
+
+                    bg_audio = np.concatenate(bg_audio)
+                    bg_audio = bg_audio[:samples_num_samples]
```

Comment:
Make this a function `repeat_until_len` or something like that. The whole `randomize_parameters` will be more readable.